src/app/containers/NewSignupContainer/AccountStep/AccountForm.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Input, PasswordInput, PrimaryButton, Field, useApi, Row, Label, EmailInput, Block } from 'react-components';
+import { c } from 'ttag';
+import { queryCheckUsernameAvailability } from 'proton-shared/lib/api/user';
+
+const AccountForm = ({ onSubmit }) => {
+    const api = useApi();
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [passwordConfirmation, setPasswordConfirmation] = useState('');
+    const [email, setEmail] = useState('');
+    const [usernameError, setUsernameError] = useState();
+
+    const handleChangeUsername = ({ target }) => {
+        if (usernameError) {
+            setUsernameError(null);
+        }
+        setUsername(target.value);
+    };
+
+    const handleChangePassword = ({ target }) => setPassword(target.value);
+    const handleChangePasswordConfirmation = ({ target }) => setPasswordConfirmation(target.value);
+    const handleChangeEmail = ({ target }) => setEmail(target.value);
```

Comment:
would be nice to have a hook for basic input handling e.g.
```
const [value, handleValueChange] = useInput('default value');
const [checked, handleCheckboxClick] = useCheckbox(false);
```
```

Reply:
yep, at least something would be nice.
```
<input onChange={asString(handleChange)} />
```